Lib/posixpath.py

```diff
@@ -552,8 +552,10 @@ def commonpath(paths):
                 common = s1[:i]
                 break
 
-        prefix = sep if isabs else sep[:0]
         return prefix + sep.join(common)
     except (TypeError, AttributeError):
-        genericpath._check_arg_types('commonpath', *paths)
+        genericpath._check_arg_types('commonpath', *(
+            # Can't use paths, can be an iterable
+            root + tail for root, tail in zip(roots, paths)
```

Comment:
`_check_arg_types` only uses types of arguments, so passing `*paths` should be OK.

Reply:
You're right, I thought I had to pass the original paths. This also simplies the diff.